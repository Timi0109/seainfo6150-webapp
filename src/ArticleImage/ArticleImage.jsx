import { isString } from "lodash";
import React from "react";
import styles from "./ArticleImage.module.css";
import PropTypes from "prop-types";

const ArticleImage = ({url, title}) =>{
    return (
        <div>
            <img src={url} ></img>
           <h2>{title}</h2>
        </div>

    )
}

ArticleImage.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  };

export default ArticleImage