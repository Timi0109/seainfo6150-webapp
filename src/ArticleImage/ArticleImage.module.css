img{
    margin-top: 10px;
    width: 140px;
    height: 100px;
    border-radius: 90%;
    border: 8px solid red;
}

h2{
    color: rgb(197, 10, 10);
    text-decoration: underline;
}