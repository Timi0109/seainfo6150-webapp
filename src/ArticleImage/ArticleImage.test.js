import React from "react";
import ArticleImage from "./ArticleImage";

describe("FunctionalComponent tests", () => {
  it("renders correctly when passed a name prop", () => {
    const { container } = render(<ArticleImage url={"https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"} title={"The Statue of Liberty's torch heads to new museum"}/>);
    expect(container).toMatchSnapshot();
  });
});
