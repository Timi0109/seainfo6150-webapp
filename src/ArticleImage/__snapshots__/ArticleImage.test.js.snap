// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalComponent tests renders correctly 1`] = `
<div>
  <div>
    <img />
    <h2 />
  </div>
</div>
`;

exports[`FunctionalComponent tests renders correctly when passed a name prop 1`] = `
<div>
  <div>
    <img
      src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
    />
    <h2>
      The Statue of Liberty's torch heads to new museum
    </h2>
  </div>
</div>
`;
